nautobot/dcim/api/serializers.py

```diff
@@ -743,15 +502,26 @@ def validate(self, data):
 
         return data
 
-    @extend_schema_field(NestedDeviceSerializer)
+    @extend_schema_field(str)
     def get_parent_device(self, obj):
         try:
             device_bay = obj.parent_bay
         except DeviceBay.DoesNotExist:
             return None
-        context = {"request": self.context["request"]}
-        data = NestedDeviceSerializer(instance=device_bay.device, context=context).data
-        data["device_bay"] = NestedDeviceBaySerializer(instance=device_bay, context=context).data
+        depth = int(self.context.get("depth", 0))
```

Comment:
I think we should keep the `depth = int(self.context.get("depth", 0))` here only because we need `self.Meta.depth` to be 1 for our OpenAPISchema. We should base our depth off of the query parameter depth if that's available, otherwise default to 0. Same comment for similar feedback regarding `self.Meta.depth` vs `self.context.get("depth", 0)`.

Reply:
I'll be honest, I still don't fully understand this. :-) I see where `Meta.depth` is decremented when we create a nested serializer, but I don't see where `context["depth"]` does this. So it seems like with this code as-is, a query with `depth=1` would result in the Device serializer rendering with depth 1, the nested device-bay serializer rendering with depth 1, its nested device serializer rendering with depth 1, its own nested device-bay serializer rendering with depth 1, etc. to infinite depth. But obviously that isn't happening... so what am I missing? :-)